src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -5070,11 +5070,7 @@ internal static void InitializeSockets()
                 {
                     if (!s_initialized)
                     {
-                        // TODO (#7849): this call is not required for *NIX and should be avoided during PAL design.
-
-                        // Ensure that WSAStartup has been called once per process.  
-                        // The System.Net.NameResolution contract is responsible with the initialization.
-                        Dns.GetHostName();
+                        SocketPal.Initialize();
```

Comment:
I considered that, but I find partial methods to be a bit confusing due to the unevenness across the PAL implementations, and since especially in this case it's only invoked once (so throughput isn't an issue) and since there's only one call site (so IL size isn't an issue), seemed fine to have the empty method defined in the Unix.cs file. And of course even with the one invocation, it should be an obvious candidate for inlining :smile: